src/os/bluestore/KernelDevice.cc

```diff
@@ -520,7 +520,7 @@ void KernelDevice::aio_submit(IOContext *ioc)
   void *priv = static_cast<void*>(ioc);
   int r, retries = 0;
   r = aio_queue.submit_batch(ioc->running_aios.begin(), e, 
-			     ioc->num_running.load(), priv, &retries);
+			     pending, priv, &retries);
```

Comment:
If I understand correct, you mean some IOs will be submit more than ones in old code. Right?

Reply:
i am not mean that, i mean there are some memory waste in struct iocb *piocb[aios_size] in submit_batch funtion. 
ios were submitted from running_aios.begin() to e, no old ios were submitted. but the total io size should be pending size, but it passed num_running.load() as aios_size to aio_submit(), num_running.load >=pending, num_running.load() may be very large, so it will make  memory waste, am i describe clear?

Reply:
clear now, thanks.

Reply:
do you really meet this case during test?